pac4j-http/src/main/java/org/pac4j/http/credentials/authenticator/DigestAuthenticator.java

```diff
@@ -0,0 +1,15 @@
+package org.pac4j.http.credentials.authenticator;
+
+import org.pac4j.http.credentials.DigestCredentials;
+import org.pac4j.http.credentials.UsernamePasswordCredentials;
+
+/**
+ * This interface represents the contract to validate digest credentials.
+ *
+ * @author Mircea Carasel
+ */
+public interface DigestAuthenticator extends Authenticator<DigestCredentials> {
```

Comment:
Not sure I can do this because DirectDigestAuthClient extends DirectHttpClient, which has a validate call, after digest Authorization token is parsed and data extracted:

    @Override
    public C getCredentials(WebContext context) throws RequiresHttpAction {
        init(context);
        try {
            C credentials = this.extractor.extract(context);
            if (credentials == null) {
                return null;
            }
            getAuthenticator().validate(credentials);
            return credentials;
        } catch (CredentialsException e) {
            logger.error("Failed to retrieve or validate credentials", e);
        }
        return null;
    }

Here is a indirect cast to DigestCredentials (the generic class C becomes DigestCredentials in DirectDigestAuthClient